wgpu-native/src/event_loop.rs

```diff
@@ -0,0 +1,66 @@
+use crate::{
+    hub::{Global, Token},
+    instance::RequestAdapterCallback,
+    id::{AdapterId, EventLoopId}
+};
+#[cfg(feature = "local")]
+use crate::{hub::GLOBAL, id::Output};
+
+#[derive(Debug)]
+pub struct EventLoop {
+    scheduled: Vec<Event>,
+}
+
+impl EventLoop {
+    pub(crate) fn schedule(&mut self, event: Event) {
+        self.scheduled.push(event);
+    }
+
+    fn process(&mut self) {
+        for event in self.scheduled.drain(..) {
+            match event {
+                Event::RequestAdapterCallback(callback, ref adapter_id, userdata) => {
+                    callback(adapter_id as *const _, userdata);
+                }
+            }
+        }
+    }
+}
+
+#[cfg(feature = "local")]
+#[no_mangle]
+pub extern "C" fn wgpu_create_event_loop() -> EventLoopId {
+    create_event_loop(&*GLOBAL)
+}
+
+#[cfg(feature = "local")]
+pub fn create_event_loop(global: &Global) -> Output<EventLoopId> {
+    let event_loop = EventLoop {
+        scheduled: Vec::new()
+    };
+    let mut token = Token::root();
+    global
+        .event_loops
```

Comment:
do we have to go through the global at all here? Perhaps, instead we should just return a pointer to the boxed structure and receive it where appropriate?

Reply:
I wasn't sure whether we need to track the event loop, i.e. for validation or some other use internally.

Relatedly I was also wondering whether the event loop should become bound to the adapter or device so that we only have to provide it once. For example, if it was pass into `request_adapter` and it becomes bound, then we could attach the event loop to anything created from that adapter (device, resources, etc.).

Reply:
Alright, I'm avoiding the global registry for now until we know it's needed